Implement Task Details page

yamin-faruque-sazim opened the pull request:
**Description of Changes**

- Refactored code to improve organization and reusability by consolidating state management and actions into the useTaskOperations hook
- Added TaskDetails component and page for viewing and editing individual tasks with detailed information
-  Slightly changed the appearance of the app 

https://github.com/user-attachments/assets/350268ea-4e82-4c7a-a35c-f0f0dd67264f

yamin-faruque-sazim committed:
FEAT(TaskDetails) -> Added TaskDetails component and page for viewing and editing individual tasks
CHORE(Refactor) -> Refactored task management logic into useTaskOperations hook and adjusted app appearance

ishmam-iqbal-sazim commented on modules/HomePage/components/TaskList/Tasklist.module.css:
```diff
@@ -0,0 +1,46 @@
+/* Assuming you are reusing TaskDetails.module.css, just add or adjust styles here if needed */
```
Should remove the comments in all the files.

ragib-hossain-sazim replied:
Please remove the comments

ragib-hossain-sazim commented on shared/utils/usetaskOperations.ts:
`usetaskOperations.ts` => `useTaskOperations.ts`

ragib-hossain-sazim commented on modules/HomePage/components/TaskList/TaskList.tsx:
```diff
@@ -104,12 +105,45 @@ const TaskList: React.FC<TaskListProps> = ({
+          className={classes.taskContainer}
         >
+          <Link
+            style={{ textDecoration: "none", color: "inherit" }}
```
please don't inline styles. Move the styles in the corresponding styles folder.

ragib-hossain-sazim commented on modules/TaskDetailsPage/TaskDetails.tsx:
```diff
@@ -0,0 +1,200 @@
+              }
+            }}
+            data={[
+              { value: "1", label: "High" },
```
define a constant for this options (for e.g 
```
export const PRIORITY_OPTIONS = [
{},
{},
{}
];
```

) in constants file and then use it here. Look into nextJS file/folder structure doc on where to define the constants file.

ragib-hossain-sazim commented on shared/utils/usetaskOperations.ts:
```diff
@@ -0,0 +1,157 @@
+    });
+  };
+
+  const applyFilter = (tasks: Task[], filter: string): Task[] => {
```
This method is doing a lot of stuff together which makes it very difficult to read. Could you please simplify this method a bit. I suggest seperating 3 different filters in 3 different methods and then adjust the caller of this method according to the change.
for e.g: 

filterByStatus
filterByPriority
filterByDueDate

yamin-faruque-sazim committed:
CHORE(FTDA): remove comments
CHORE(FTDA): rename custom hook file
CHORE(FTDA): use a constants file for priority options data
CHORE(FTDA): simplify and breakdown the filters function into smaller functions

yamin-faruque-sazim commented:
Addressed the comments

ishmam-iqbal-sazim commented on modules/HomePage/components/TaskList/TaskList.tsx:
```diff
@@ -13,6 +14,8 @@ import {
 import { useDisclosure } from "@mantine/hooks";
 
 import { Task } from "../../types/Task";
+import classes from "./TaskList.module.css";
+import { PRIORITY_OPTIONS } from "@/shared/constants/taskContstants";
```
@ imports should be above relative imports

ishmam-iqbal-sazim reviewed (approved):
Left a comment
